src/tests/function/left_right.rs

```diff
@@ -0,0 +1,179 @@
+        ),
+        (
+            r#"INSERT INTO SingleItem VALUES (0)"#,
+            Ok(Payload::Insert(1)),
+        ),
+        (
+            "CREATE TABLE NullName (name TEXT NULL)",
+            Ok(Payload::Create),
+        ),
+        (
+            r#"INSERT INTO NullName VALUES (NULL)"#,
+            Ok(Payload::Insert(1)),
+        ),
+        (
+            "CREATE TABLE NullNumber (number INTEGER NULL)",
+            Ok(Payload::Create),
+        ),
+        (
+            r#"INSERT INTO NullNumber VALUES (NULL)"#,
+            Ok(Payload::Insert(1)),
+        ),
+        (
+            "CREATE TABLE NullableName (name TEXT NULL)",
+            Ok(Payload::Create),
+        ),
+        (
+            r#"INSERT INTO NullableName VALUES ('name')"#,
+            Ok(Payload::Insert(1)),
+        ),
+        (
+            r#"SELECT LEFT(name, 3) AS test FROM Item"#,
+            Ok(select!(
+                "test"
+                Str;
+                "Blo".to_owned();
+                "B".to_owned();
+                "Ste".to_owned()
+            )),
+        ),
+        (
+            r#"SELECT RIGHT(name, 10) AS test FROM Item"#,
+            Ok(select!(
+                "test"
+                Str;
+                "op mc blee".to_owned();
+                "B".to_owned();
+                "d$ folken!".to_owned()
+            )),
+        ),
+        // TODO Concatenation
+        /*(
+            r#"SELECT LEFT((name + 'bobbert'), 10) AS test FROM Item"#,
+            Ok(select!(
+                "test"
+                Str;
+                "Blop mc blee".to_owned();
+                "Bbobbert".to_owned();
+                "Steven the".to_owned()
+            )),
+        ),*/
+        (
+            r#"SELECT LEFT('blue', 10) AS test FROM SingleItem"#,
+            Ok(select!(
+                "test"
+                Str;
+                "blue".to_owned()
+            )),
+        ),
+        (
+            r#"SELECT LEFT("blunder", 3) AS test FROM SingleItem"#,
+            Ok(select!(
+                "test"
+                Str;
+                "blu".to_owned()
+            )),
+        ),
+        (
+            r#"SELECT LEFT(name, 3) AS test FROM NullName"#,
+            Ok(select!(
+                "test"
+                OptStr;
+                None
+            )),
+        ),
+        (
+            r#"SELECT LEFT('Words', number) AS test FROM NullNumber"#,
+            Ok(select!(
+                "test"
+                OptStr;
+                None
+            )),
+        ),
+        (
+            r#"SELECT LEFT(name, number) AS test FROM NullNumber INNER JOIN NullName ON 1 = 1"#,
+            Ok(select!(
+                "test"
+                OptStr;
+                None
+            )),
+        ),
+        (
+            r#"SELECT LEFT(name, 1) AS test FROM NullableName"#,
+            Ok(select!(
+                "test"
+                Str;
+                "n".to_owned()
+            )),
```

Comment:
The last one, I saw that you changed to use `OptStr`, and this would be also good to use `OptStr`.
If it doesn't, then
`OptStr(Some(..))` becomes `Str` and `OptStr(None)` becomes `OptStr`.
Then column type can be different for each row.
Thought current tests only have 1 row, but using 2 rows with `OptStr(Some("a".to_owned()))` and `OptStr(None)`, we can break the column type consistency.

Reply:
Good point, just left them there.